samples/backend/python/quart/__init__.py

```diff
@@ -0,0 +1,181 @@
+import base64
+import json
+import os
+import re
+from dotenv import load_dotenv
+from typing import Optional, Tuple
+from openai import AsyncAzureOpenAI
+from azure.identity import DefaultAzureCredential, get_bearer_token_provider
+from quart import Quart, request, jsonify, stream_with_context
+from model import (
+    AIChatRequest,
+    AIChatCompletion,
+    AIChatMessage,
+    AIChatRole,
+    AIChatFile,
+    AIChatError,
+    AIChatCompletionDelta,
+    AIChatMessageDelta,
+)
+
+load_dotenv()
+
+PORT = os.getenv("PORT", 3000)
+AZURE_OPENAI_ENDPOINT = os.getenv("AZURE_OPENAI_ENDPOINT")
+AZURE_OPENAI_DEPLOYMENT = os.getenv("AZURE_OPENAI_DEPLOYMENT")
+AZURE_OPENAI_API_VERSION = os.getenv("AZURE_OPENAI_API_VERSION")
+
+token_provider = get_bearer_token_provider(
+    DefaultAzureCredential(), "https://cognitiveservices.azure.com/.default"
+)
+client = AsyncAzureOpenAI(
+    api_version=AZURE_OPENAI_API_VERSION,
+    azure_endpoint=AZURE_OPENAI_ENDPOINT,
+    azure_ad_token_provider=token_provider,
+)
+
+app = Quart(__name__)
+
+
+def run() -> None:
+    app.run(port=PORT)
+
+
+def get_file_position(file_key: str) -> Optional[Tuple[int, int, str]]:
+    match = re.match(r"messages\[(\d+)\]\.files\[(\d+)\]", file_key)
+    if match:
+        message_index, file_index = map(int, match.groups())
+        return message_index, file_index, file_key
+    raise ValueError(f"Invalid file name: {file_key}")
+
+
+def reconstruct_multipart_request(form: dict, files: dict):
```

Comment:
Is the idea of this function that youre matching multiple files to multiple messages? A docstring might be helpful.
This seems to be a lot more code than my current code, so I want to make sure I understand the benefit of this more complicated approach.

Reply:
The get_file_position function could also use a docstring/explanation

Reply:
This is matching multiple files to multiple messages. I can document a bit more :)